modules/mysql/testcontainers/mysql/__init__.py

```diff
@@ -104,8 +106,9 @@ def _connect(self) -> None:
         )
 
     def get_connection_url(self) -> str:
+        dialect = "mysql" if self.dialect is None else f"mysql+{self.dialect}"
```

Comment:
I think you should put this simply in the `init` when you set `self.dialect`. 

It would effectively become:
```python
self.dialect = dialect or environ.get("MYSQL_DIALECT") or "mysql"
```

also I think `dialect="mysql+asdf"` makes more sense than `dialect="asdf"`

Reply:
I think that dialect is a specific extension that comes after "mysql". Why repeat “mysql” everywhere if we already know that we are connecting to mysql?

Reply:
@totallyzen Not that this is an argument, but it was interesting to look at the [current implementation of the postgresql container](https://github.com/testcontainers/testcontainers-python/blob/main/modules/postgres/testcontainers/postgres/__init__.py#L55), and it is done according to the principle that I proposed

Reply:
> and it is done according to the principle that I proposed. what do you think?

sure I don't mind - maybe then validate the dialect _does not_ start with `mysql` (raise a `ValueError`) then I'm happy :)

Reply:
ready to make this check what proposed @balint-backmaker if you approve it @totallyzen

Reply:
oops, sorry for the mix - commented from the company account and not my personal :sweat_smile:

Reply:
@totallyzen did it in last commit

Reply:
@totallyzen ping)